app/components/WhitelistedDApps/index.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { normalizeUrl } from '../../utils/helpers'
+
+import { addWhitelistedDApp, deleteWhitelistedDApp } from '../../actions/whitelistedDApps'
+
+class WhitelistedDApps extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      newDApp: '',
+      errorMessage: '',
+    }
+  }
+
+  updateNewDApp(dApp) {
+    this.setState({ newDApp: dApp })
+  }
+
+  validateDApp(dApp) {
+    if (!dApp) {
+      this.setState({ errorMessage: 'Invalid DApp' })
```

Comment:
If you use the state for knowing if the component is in "error" mode, you do not need to return a boolean. Just let the component to be re-rendered and in the render method introduce logic. Something like:
```
public render() {
   <div>
     { errorMessage && <div> Incredible error </div> }
     <div>rest of html</div>
   </div>
}
```